### PR TITLE
 Initial refactor from alloy

### PR DESCRIPTION
**Pull request by LinuxBozo**
Refactors alloy SAML enabled Cognito module in the style of Truss.

### CODE REVIEW

**Comment by LinuxBozo** [comment]
terratest result

```terratest
--- PASS: TestCognitoCreation (1377.32s)
PASS
ok  	github.com/trussworks/terraform-aws-saml-cognito/test	1377.658s
```

**Comment by LinuxBozo** [comment]
Tests currently fail circleci validate, blocked by https://github.com/trussworks/legendary-waddle-dev/pull/144

**Comment by LinuxBozo** [comment]
... And we're unblocked and good in CI.

File: examples/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
+locals {
+  zone_name                = "infra-test.truss.coffee"
+  idp_redirect_binding_uri = "https://samltest.id/idp/profile/SAML2/Redirect/SSO"
```
Comment (dynamike) [inline examples/simple/main.tf:3]: TIL.

**Review by dynamike — APPROVED** [review]

**Review by eeeady — APPROVED** [review]
LGTM! Thank you for pulling this out.
(sorry... nearly forgot to approve after I ran tests earlier)

**Pull request merged**